containers/layout/Layout.tsx

```diff
@@ -2,39 +2,77 @@ import React from 'react';
 import Container from '@material-ui/core/Container';
 import { I18n, TFunction } from 'next-i18next';
 import Head from 'next/head';
-import { getTitleString } from 'lib/utils/strings';
 import Footer from 'components/footer/Footer';
 import useStyles from './LayoutStyles';
 import { Box, Typography } from '@material-ui/core';
 import OptionToggler from 'components/option-toggler/OptionToggler';
 import { Button } from 'components';
 import { useAuth } from 'hooks/useAuth';
 import { useRouter } from 'next/router';
-import { FirebaseError } from 'lib/types/auth';
 import Error from 'components/error';
+import { withTranslation } from 'i18n.config';
+import { layoutErrorVar } from 'apollo/cache';
+import { useReactiveVar } from '@apollo/client';
+import { getTitleString, getTitleStringFromPathname } from 'lib/utils/strings';
+import { gql, useQuery } from '@apollo/client';
+import { getShortName } from 'lib/utils/strings';
+
+const GET_TALENT_NAME = gql`
```

Comment:
Hey Tobi. Sorry, I know I suggested this myself, but I am not really happy with this solution of having another graphql query in the layout, even though the info should be available from the component. Unfortunatly, I forgot the reason why we cannot just set the custom title for talents pages on the talents profile page itself (with layoutVar({title: [name from graphql query] || 'Talent Profile'}) or whatever).This should first return 'Talent Profile', and then, when the query is resolved, it should return the name from that query (or however we want to format the title). I know you said something about this is not working, but I don't recall what. Could you remind me. (Sorry about this, if I find the time, I'll download your branch and try it out myself).

Reply:
Oh. Everything else is fine by me, btw.

Reply:
@atsteinberg I am looking at it right now.
The reason why we decided not to set the title on the talents page is that the setting of the title triggers another component re-render and we end up in an endless loop.
This was the problem when used React Context. But I actually also tried it out with the reactive variable and I had the same problem.
I am double-checking now.

Reply:
ufffffff when I tried this out before, I was not using the proper `useReactiveVar` hook.
I just did a quick test and it looks like it's working with setting the title directly from the component and not ending in an endless loop.
I will work on that.

Reply:
OK. I made this work without another graphql query. Instead of using a layoutErrorVar, use a more general one that can take error, title and heading. On the talents page, use this useEffect:
```
  useEffect(() => {
    if (data) {
      const title = `profile | ${data.getTalentById.basicInfo.fullName}`;
      const heading = `${t('pages.profile.greeting')}, ${
        data.getTalentById.basicInfo.name.firstName
      }!`;
      layoutVar({ title, heading });
    }
  }, [data, t]);
```

In Layout get title and heading with useReactiveVar(layoutVar) and set titleToDisplay and headingToDisplay (or however you want to name it) as follows:
```
const titleToDisplay = title || getTitleFromPathname(pathname);
const heading = heading || '';
```

Saves the extra query.

Reply:
or better still: only pass the name info to layout and do the formatting of title and heading there.

Reply:
- One thing we have to consider then: We have to make sure to reset the `title` from our `layoutVar` to an empty string in the callback of the `useEffect` on the profile page.
- not sure if passing the name info to layout and doing the formatting there is better. If we get the `title` from several different pages, it's better to just receive the `title` that we can display. The formatting is then specific to each page (where we want to set a specific `title`)